### DISCUSSION TITLE
PyScript OpenCV -- Blank Screen

### DISCUSSION THREAD

**Discussion by guruvishnuvardan** [pyscript/pyscript#2032] [Q&A]
I am attempting to load an opencv-python functionality to HTML using pyscript, but I am only getting blank browser screen after the messages - "Loading Runtime ... Runtime Created ... Initializing Components", as attached in the screenshot

Please find the python code and HTML for your review, Can you please let me know, if anything is wrong.

```python
import cv2
import os
import numpy as np
import inotify.adapters
import glob
import time
from pathlib import Path
from rembg import remove, new_session
import glob
from pyscript import display


session = new_session()
edgeTotal = 0.0
for file in Path('/home/kali/Desktop/TVS/REMBG/').glob('*.png'):
    input_path = str(file)
    #output_path = str('/home/kali/Downloads/REMBG/' + (file.stem + ".out.png"))

    with open(input_path, 'rb') as i:
        input = i.read()
        #print(input)
        input_img = cv2.imread(input_path)
        #img = cv2.resize(input_img, (500, 500))
        img_gray = cv2.cvtColor(input_img, cv2.COLOR_BGR2GRAY)
        # Blur the image for better edge detection
        img_blur = cv2.GaussianBlur(img_gray, (3, 3), 0)
        gray = cv2.cvtColor(input_img, cv2.COLOR_BGR2GRAY)
        edged = cv2.Canny(gray, 30, 200)
        edgedRes = 0
        #edgeTotal = 0.0
        edgedRes = np.sum(edged)

        edgeTotal += edgedRes
        print("summa")
        print(edgeTotal)
        display(edgeTotal, target="mpl")
```

![image](https://github.com/pyscript/pyscript/assets/31349523/d9204207-d0c3-47c0-a742-3e6370ac993d)

![6lqy6](https://github.com/pyscript/pyscript/assets/31349523/2f988cae-7fe2-4e26-a66a-d289300d456a)

Thanks
Guru

**Comment by JeffersGlass** [comment]
PyScript has come a long way since the `alpha` release from 2022 - I would start by updating to the newest version and syntax. See [the documentation](https://pyscript.github.io/docs/2024.4.1/beginning-pyscript/#indexhtml) for the scaffolding of what using modern PyScript looks like.

**Reply by guruvishnuvardan** [reply]
@JeffersGlass  - Thanks for your quick response. I have seen the PyScript Example Code(s) and constructed my  python code and HTML.

Can you please help me by sharing a sample where python and opencv are executed in the browser.

Thanks
Guru

**Reply by JeffersGlass** [reply]
You're using a very old version of PyScript, and there have likely been improvements and changes both to PyScript and opencv in that time. In particular, the syntax for Py-config is entirely different now, as is the link to PyScript itself. So one of the reasons it's not working is that you're not actually installing opencv, because the `py-env` tag no longer exists.a

**Reply by JeffersGlass** [reply]
As a place to start, I don't know what the `inotify` or `rembg` packages are, but they probably want to be included in your configuration.

**Reply by guruvishnuvardan** [reply]
@JeffersGlass  -- Thanks for your advise. I will change the Code , as per the samples from Pyscript website and I will also include the TOML Configuration file.

rembg is a package to remove background of an image and inotify is used for file handling in python.

I wil post the results shortly.

**Reply by guruvishnuvardan** [reply]
@JeffersGlass  - Please note,I have updated the python code and HTML, but I am encountering the same result(blank screen) and I tried to execute the python code using the linux terminal, I have encountered the following error:

 from pyscript import display
ImportError: cannot import name 'display' from 'pyscript' (/usr/local/lib/python3.11/dist-packages/pyscript/__init__.py)

Now, I am printing the value using print statement, instead of display from pyscript

Please find the attached code, HTML and config file

`import cv2
import os
import numpy as np
import glob
import time
from pathlib import Path
#from pyscript import display

#session = new_session()
edgeTotal = 0.0
for file in Path('/home/kali/Desktop/TVS/REMBG/').glob('*.png'):
    input_path = str(file)
    #output_path = str('/home/kali/Downloads/REMBG/' + (file.stem + ".out.png"))

    with open(input_path, 'rb') as i:
        input = i.read()
        #print(input)
        input_img = cv2.imread(input_path)
        #img = cv2.resize(input_img, (500, 500))
        img_gray = cv2.cvtColor(input_img, cv2.COLOR_BGR2GRAY)
        # Blur the image for better edge detection
        img_blur = cv2.GaussianBlur(img_gray, (3, 3), 0)
        gray = cv2.cvtColor(input_img, cv2.COLOR_BGR2GRAY)
        edged = cv2.Canny(gray, 30, 200)
        edgedRes = 0
        #edgeTotal = 0.0
        edgedRes = np.sum(edged)
        edgeTotal += edgedRes
        #print(edge)
        print(edgeTotal)`







Config FIle:

name = "Edge"
description = "OpenCV Edge Detection"
packages = ["opencv-python", "numpy", "pathlib"]


![Screenshot_2024-04-18_22_40_35](https://github.com/pyscript/pyscript/assets/31349523/a7311e4f-d9e8-4ceb-8275-7a5e89663e40)

Can you please let me know, if there is any mistake made again.

Thanks
GUru

**Reply by WebReflection** [reply]
> for file in Path('/home/kali/Desktop/TVS/REMBG/').glob('*.png'):

you can't load files in PyScript from your local hard drive out of the box, you need to upload files or attach these to the virtual File System then grab these from there but also save these from there.

The `pyscript` namespace doesn't exist in pyodide out of the box so testing in Linux terminal won't bring you far neither ... they are the same pyodide but one has full access to your File System, the other one running on the Web can't directly access that due security reasons of the Web.

See this example on how to upload and download and put your logic with your packages in between: https://pyscript.recipes/2024.1.1/basic/file-upload/

**Reply by guruvishnuvardan** [reply]
@WebReflection  -- Thanks for your response. I will work as per your advise.

**Comment by guruvishnuvardan** [comment]
Closing this Issue.
